### ISSUE TITLE
.== not defined for tuples

### ISSUE THREAD

**Issue by jiahao** [JuliaLang/julia#4827]
Having `.==` work on tuples would simplify a lot of dimensions checking in the linear algebra routines.

```
julia> A=randn(10,10);

julia> any(size(A).==10)
ERROR: no method .==((Int64,Int64),Int64)

julia> all([size(A)...].==10)
true
```

**Comment by IainNZ** [comment]
@jiahao is this still a desired feature?

**Comment by jiahao** [comment]
yes

**Comment by mbauman** [comment]
Implemented by #17623, which has also answered how to handle `tuple .== array`.  It's also super-efficient when using same-length tuples, but comparisons against scalars could probably be optimized further if you want to use this for linear algebra dimension checks.
